batch/batch/worker/worker.py

```diff
@@ -474,11 +474,18 @@ async def localize_rootfs():
     async def run_until_done_or_deleted(self, f: Callable[[], Awaitable[Any]]):
         step = asyncio.ensure_future(f())
         deleted = asyncio.ensure_future(self.deleted_event.wait())
-        await asyncio.wait([deleted, step], return_when=asyncio.FIRST_COMPLETED)
-        if deleted.done():
-            raise JobDeletedError()
-        assert step.done()
-        return step.result()
+        try:
+            await asyncio.wait([deleted, step], return_when=asyncio.FIRST_COMPLETED)
+            if deleted.done():
+                raise JobDeletedError()
+            assert step.done()
+            return step.result()
+        finally:
+            for t in (step, deleted):
+                if t.done() and t.exception():
+                    log.exception(t.exception())
```

Comment:
This will log an exception for user tasks failing. Do we want that? Or do we catch whether it's a user error in an upstream function call?

Reply:
Good point, I can add a check here for `user_error`.

Reply:
I don't think we have a robust way of guaranteeing that upstream calls check or not, or propagate or not. Long term I think there's a better way of splitting up the error cases further but that's for another time I suppose.